### ISSUE TITLE
Optimize Event Subscribers based on actual changed data

### ISSUE THREAD

**Issue by DiegoPino** [esmero/strawberryfield#117]
@giancarlobi this is to discuss with you.

Was thinking that we can actually skip a lot of subscribers if we check if there are actually any changes on the SBF s or not? This applies of course to Flavours/Processors too.
We can compare any previous data (using $entity->original v/s $entity) and if the SBF are == then no need to process right?)
And if there are no changes to files (e.g there are changes to metadata but all the fields that hold files are ==), we can save a lot of FOREACH and skip file classification checks.

Do you see any possible problems with this approach?  Thanks!

**Comment by giancarlobi** [comment]
@DiegoPino To approach this I'm thinking in an ADO as made by three main components: SBF-JSON (metadata), files (images, zipped, miniocr, ...) and Solr indexed docs. We have to take into account all of them.
So, we can skip a step if all expected results will be the same as the current ones, again SBF-JSON, files and Solr. 
How can we evaluate if we can skip a step, that is if results are already updated? 
Well, we have to suppose that if we input A to a subcriber/flavor/processor, it always and only returns B, so we can compare inputs (last and current) and if they are the same then we can skip.
This means that for each step, we want check to skip or not, we must compare all inputs from which the result depend on. This can be achieved if all inputs are available, that is can be accessed by the code to be compared.
In conclusion, decide if skip or not is not so simple but I think that in anyway it's a good approach to save machine time (i.e. check if we don't need to make ocr again is really less time consuming than run tesseract, as you know).
Sorry, friend, if my answer is a little bit "philosophical" and not concrete as you probably expected.

**Comment by DiegoPino** [comment]
Closing as won't implement. The checks are more expensive than the processing
